app/Services/Model/Imp/SummitTicketTypeService.php

```diff
@@ -350,6 +369,32 @@ public function getAllowedTicketTypes(Summit $summit, Member $member): array
                 )
             );
 
+            $promocode = null;
+
+            if (!is_null($filter)) {
+                $element = $filter->getUniqueFilter('promo_code');
```

Comment:
filter logic should live on controller , this layer just need string? promo_code

Reply:
done